app/models/rule.rb

```diff
@@ -69,9 +77,12 @@ def canonical?
     (profiles & Profile.canonical).any?
   end
 
-  def self.from_openscap_parser(op_rule, benchmark_id: nil)
-    rule = find_or_initialize_by(ref_id: op_rule.id,
-                                 benchmark_id: benchmark_id)
+  def self.from_openscap_parser(op_rule, benchmark_id: nil, group_id: nil)
+    rule = create_with(group_id: group_id).find_or_initialize_by(ref_id: op_rule.id,
+                                                                 benchmark_id: benchmark_id)
+    if rule.group_id != group_id
+      rule.update!(group_id: group_id)
+    end
```

Comment:
This is heavily inefficient if ran one-by-one :thinking: should be implemented in the import statement somehow. I have some ideas that we can discuss.

Reply:
Yeah, I'd like to hear your ideas! The `update!` would only run if the `group_id` changes which might not be too common? Unless we expect the group_id for a rule to change a lot?